### ISSUE TITLE
iohook is looking for a new maintainer !

### ISSUE THREAD

**Issue by Djiit** [wilix-team/iohook#291]
Hey everyone,

2020 has been a strange year, and 2021 is starting to look pretty much the same. tbh I don't think I'll have more time to dedicate to iohook. Updates have become really rare, and it is only thanks to this awesome community that we recently added support for modern versions of Node and Electron.

So, that's it, I don't have anymore free time to dedicate to this project, and I don't use it anymore on any personal project, so I'm looking for a motivated maintainer to help with ongoing updates, triage the issues, fix bugs, etc...

I'll stick around, for sure, but we need a new person in charge.

Cheers

**Comment by WilixLead** [comment]
Hi! Looks like I can return to maintaining this module

**Comment by Djiit** [comment]
Well, that was unexcepted -- but very welcome :)

**Comment by WilixLead** [comment]
:D At this moment we have partially free hands in company. Just need some time for review current state and new PR's.
We try write some tests and add some automatisations for creating prebuild's for new Node/Electron versions

**Comment by Djiit** [comment]
Awesome !
We recently switch to node-gyp instead of cmake-js, so yeah the project would need a revamped automated CI (maybe using more recent versions of prebuilt for instance), plus some good tests, you are right :). Let me know if you want to have a quick chat. I'm keeping this issue open a few days to see if anyone wants to join the fun.

**Comment by WilixLead** [comment]
Great! I just need remember how CI upload new builds to NPM :)

**Comment by tcardlab** [comment]
> :D At this moment we have partially free hands in company. Just need some time for review current state and new PR's.
> We try write some tests and add some automatizations for creating prebuild's for new Node/Electron versions

@WilixLead, if you are reviewing the current state and updating prebuilds, could you take a look into mac support?
There is a long-standing [mac issue](https://github.com/wilix-team/iohook/issues/124#issuecomment-451226505) that is still causing confusion ( see #295 ).

There is a [hack](https://github.com/wilix-team/iohook/issues/124#issuecomment-513026388) that can be implemented ([example](https://github.com/tcardlab/iohook)), but I wonder if it would be possible to update to libuiohook v1.2?

**Comment by marcelblum** [comment]
@tcardlab I have an experimental fork adapted to use libuiohook 1.2 here https://github.com/marcelblum/iohook you are welcome to try building from there for Mac to test if it fixes the issue. For me I have never been able to replicate #124 and I use iohook on Mac all the time, though only in Electron.

**Comment by veeramarni** [comment]
@WilixLead I still have issue to make this library work in latest MacOS, did you update libuiohook to 1.2?

**Comment by WilixLead** [comment]
@veeramarni No. It is still not updated

**Comment by TStod** [comment]
@marcelblum #124 seems to be only an issue on node and not electron. I moved iohook from my main process (running electron) to a child process (only node) and saw the exact issue mentioned. Would be interesting to se if libuiohook 1.2 update would fix the issue. @WilixLead

**Comment by ash0x0** [comment]
@WilixLead @Djiit Who's the current maintainer of this project? The last time there was any activity from any maintainer is April 19. I and others have to resort to scoped packages and forks to work with this project. If the current maintainer isn't interested anymore please indicate so the project can have a new maintainer as it's an active project.

**Comment by Djiit** [comment]
Hey, as stated before, I can no longer maintain this project. @WilixLead did you pick it up in the end ?

**Comment by WilixLead** [comment]
@Djiit Thank you for reopen. I managed to fix only a couple of mistakes, and then time is end :)

**Comment by WilixLead** [comment]
@ash0x0 you want to be maintainer?

**Comment by ash0x0** [comment]
> @ash0x0 you want to be maintainer?

Would love to. I'm already familiar with the code and I do have the time to invest in this.

**Comment by WilixLead** [comment]
Ok. I add you write access. Make iohook great again :)

**Comment by ash0x0** [comment]
> Ok. I add you write access. Make iohook great again :)

Thank you, will do my best :)

**Comment by ash0x0** [comment]
@Djiit @WilixLead I'll need access to the NPM package, someone add me as collaborator please, same username.

**Comment by WilixLead** [comment]
@ash0x0 Please check your email

**Comment by WilixLead** [comment]
Also I think the best way to publish npm package - it with CI step

**Comment by ash0x0** [comment]
> @ash0x0 Please check your email

Thank you, got it.

> Also I think the best way to publish npm package - it with CI step

@WilixLead I believe so too. You'll need to add an NPM automation token to the repo secrets under `NODE_AUTH_TOKEN` for that to work.

**Comment by ash0x0** [comment]
Also, you'll need to disable AppVeyor and travis (if enabled) integration for this repo. AppVeyor is still reporting builds as failed.

**Comment by WilixLead** [comment]
@ash0x0 Github actions already done? No need travis and appveyor?

**Comment by ash0x0** [comment]
> @ash0x0 Github actions already done? No need travis and appveyor?

@WilixLead yes, actions is doing releasing now, I've already deleted travis and appveyor configs. Once you add `NODE_AUTH_TOKEN` secret it'll publish NPM package as well instead of me doing it manually, let me know when you've added it because I can't see secrets.

**Comment by WilixLead** [comment]
@ash0x0 I add NODE_AUTH_TOKEN with token now. Try now if you can

**Comment by ash0x0** [comment]
@WilixLead Confirmed, working well. Thank you.

**Comment by giladdarshan** [comment]
Hi @ash0x0, I will be happy to help maintain and fix bugs, I will need to know how and whats the best way.

**Comment by ash0x0** [comment]
> Hi @ash0x0, I will be happy to help maintain and fix bugs, I will need to know how and whats the best way.

That'd be more than welcome. You can check the Todo milestone in issues to see what needs to be fixed. As far as contributing goes, the docs have good info on that. 
I'd say pick any issue you think you can fix and ping me there and I'll provide help.
I see you did some work on #214, maybe you can PR that. You'd need to test on linux and windows, I recommend making use of gh actions and tests. If you can add a unit test action that'd be great for all future PRs too.

**Comment by lacymorrow** [comment]
@WilixLead @marcelblum @veeramarni @TStod 
Anyone have maintenance access? This library has gone stagnant and there are some good PRs open

**Comment by marcelblum** [comment]
@lacymorrow your best bet is to use [uiohook-napi](https://github.com/SnosMe/uiohook-napi) instead, it's almost a drop-in replacement for iohook and is actively maintained and fully compatible with latest node and electron. iohook seems dead unfortunately but that's ok because it would need refactoring to be made context aware and better support latest node/electron, and uiohook-napi has already done that from scratch, using the same underlying library libUIOHook.

**Comment by ZTaer** [comment]
> @lacymorrow你最好的选择是使用[uiohook-napi](https://github.com/SnosMe/uiohook-napi)，它几乎是 iohook 的直接替代品，并且得到积极维护并与最新的节点和电子完全兼容。不幸的是，iohook 似乎已经死了，但这没关系，因为它需要重构才能感知上下文并更好地支持最新的节点/电子，而 uiohook-napi 已经使用相同的底层库 libUIOHook 从头开始​​做到了这一点。

Very user-friendly

**Comment by WilixLead** [comment]
> @WilixLead @marcelblum @veeramarni @TStod Anyone have maintenance access? This library has gone stagnant and there are some good PRs open

Unfortunately at this moment we have no active maintainers. If some one want to help - welcome :)

**Comment by lacymorrow** [comment]
I would be happy to manage the repo, respond to issues, and merge PRs if you would like to give me access. I have been using IOHook in a production app for 5 or 6 years now.

There are also a few other folks here that seem eager.

**Comment by WilixLead** [comment]
I was give you access :)

**Comment by lacymorrow** [comment]
Thank you! I'll be responsible.

**Comment by WilixLead** [comment]
I'm very sorry about current project state. Can't find any time to figure out issues and PR's. So all in your hand's now :) 
Feel free asking any questions or permissions

**Comment by lacymorrow** [comment]
No worries! I maintain my own libraries and I know how much life affects our ability to work on passion projects.

I appreciate all the hard work you've done building the library! It has enabled me to add tons of features to my apps over the years.

**Comment by lacymorrow** [comment]
PRs welcome.
